src/app/pages/rpc-stx-call-contract/components/call-contract-details.layout.tsx

```diff
@@ -0,0 +1,124 @@
+import { useAsync } from 'react-async-hook';
+
+import { PayloadType, addressToString, fetchAbi } from '@stacks/transactions';
+import { SharedComponentsSelectors } from '@tests/selectors/shared-component.selectors';
+import { Stack, styled } from 'leather-styles/jsx';
+
+import { formatContractId } from '@leather.io/stacks';
+import { AddressDisplayer, Approver } from '@leather.io/ui';
+
+import { logger } from '@shared/logger';
+
+import { useStacksExplorerLink } from '@app/common/hooks/use-stacks-explorer-link';
+import { LoadingSpinner } from '@app/components/loading-spinner';
+import { useRpcRequestContext } from '@app/features/rpc-request/rpc-request.context';
+
+import type { RpcCallContractRequest } from '../rpc-stx-call-contract-container';
+import { FunctionArgumentList } from './function-argument-list';
+
+export function CallContractDetailsLayout() {
+  const {
+    rpcRequest: { unsignedTx },
+  } = useRpcRequestContext<RpcCallContractRequest>();
+  const { handleOpenStacksTxLink } = useStacksExplorerLink();
+
+  if (unsignedTx.payload.payloadType !== PayloadType.ContractCall)
+    throw new Error('Transaction is not a contract call');
+
+  const contractAddress = addressToString(unsignedTx.payload.contractAddress);
+  const contractName = unsignedTx.payload.contractName.content;
+  const functionName = unsignedTx.payload.functionName.content;
+
+  const contractAbi = useAsync(async () => {
+    try {
+      return await fetchAbi({ contractAddress, contractName });
+    } catch (error) {
+      logger.error(`Failed to fetch contract ABI for ${contractAddress}.${contractName}`, error);
+      return;
+    }
+  }, []);
```

Comment:
Shouldn't this be a query?

Reply:
I was kinda experimenting with this function from `@stacks/transactions`. We weren't using it before, so I wasn't sure if it was the right direction. We used to actually perform the contract call to validate it, but maybe that was before this existed. I will relocate it if keep.